scout/scout_search/scout_search.cpp

```diff
@@ -123,32 +153,61 @@ void resume_mission(Mission& mission)
     }
 }
 
+void drone_status(Telemetry& telemetry, Action& action, Mission& mission)
+{
+    get_position_info(telemetry);
+    std::cout << "Loitering: " << telemetry.in_air() << std::endl;
+
+    std::pair<Mission::Result, bool> mission_finished = mission.is_mission_finished();
+    std::cout << "Mission Finished: " << mission_finished.second << std::endl;
+
+    if (telemetry.in_air() && mission_finished.second) {
+        std::cout << "Mission Finished... landing" << std::endl;
+        const auto land_result = action.land();
+        if (land_result != Action::Result::Success) {
+            std::cerr << "Landing failed: " << land_result << '\n';
+        }
+    }
+}
+
 void process_movement_command(
-    const std::string& command, Action& action, Mission& mission, Telemetry& telemetry, std::vector<Mission::MissionItem>& mission_items)
+    const std::string& command,
+    Action& action,
+    Mission& mission,
+    Telemetry& telemetry,
+    std::vector<Mission::MissionItem>& mission_items)
 {
     std::istringstream iss(command);
     std::string cmd;
     iss >> cmd;
     std::cout << "Command: " << cmd << std::endl;
 
     if (cmd == "upload") {
+        mission.set_return_to_launch_after_mission(true);
         upload_mission(mission, mission_items);
 
-    }else if (cmd == "add_waypoint") {
+    } else if (cmd == "upload_no_rtl") {
+        mission.set_return_to_launch_after_mission(false);
+        upload_mission(mission, mission_items);
+    } else if (cmd == "status") {
+        drone_status(telemetry, action, mission);
+    } else if (cmd == "add_waypoint") {
```

Comment:
why not use a switch statement?

Reply:
true it would be more efficient in this case